presto-hive/src/test/java/io/prestosql/plugin/hive/parquet/ParquetDecimalScalingTest.java

```diff
@@ -0,0 +1,328 @@
+    }
+
+
+    @DataProvider(name = "different-precisions")
+    public Object[][] differentPrecisions()
+    {
+        return new Object[][] {
+                {"decimal_10_2", false, 10, 2, ImmutableList.of(
+                        "10.01", "10", "1.23", maximumValue(10, 2), minimumValue(10, 2)),
+                        ImmutableList.of("10.01", "10.00", "1.23", maximumValue(10, 2), minimumValue(10, 2))},
+                {"decimal_10_2_fle", true, 10, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(10, 2), minimumValue(10, 2)),
+                        ImmutableList.of("10.01", "10.00", "1.23", maximumValue(10, 2), minimumValue(10, 2))},
+                {"decimal_4_2", false, 4, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(4, 2), minimumValue(4, 2)),
+                        ImmutableList.of("10.01", "10.00", "1.23", maximumValue(4, 2), minimumValue(4, 2))},
+                {"decimal_4_2_fle", true, 4, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(4, 2), minimumValue(4, 2)),
+                        ImmutableList.of("10.01", "10.00", "1.23", maximumValue(4, 2), minimumValue(4, 2))},
+                {"decimal_14_2", false, 14, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(14, 2), minimumValue(14, 2)),
+                        ImmutableList.of("10.01", "10.00", "1.23", maximumValue(14, 2), minimumValue(14, 2))},
+                {"decimal_6_3", false, 6, 3,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(6, 3), minimumValue(6, 3)),
+                        ImmutableList.of("10.010", "10.000", "1.230", maximumValue(6, 3), minimumValue(6, 3))},
+                {"decimal_6_3_fle", true, 6, 3,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(6, 3), minimumValue(6, 3)),
+                        ImmutableList.of("10.010", "10.000", "1.230", maximumValue(6, 3), minimumValue(6, 3))},
+                {"decimal_38_4", false, 38, 4,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(38, 4), minimumValue(38, 4)),
+                        ImmutableList.of("10.0100", "10.0000", "1.2300", maximumValue(38, 4), minimumValue(38, 4))}
+        };
+    }
+
+    protected void createTable(String tableName, int precision, int scale)
+    {
+        assertUpdate(format("create table %s (value decimal(%d, %d)) WITH (format = 'PARQUET')", tableName, precision, scale));
+    }
+
+    protected void dropTable(String tableName)
+    {
+        assertUpdate(format("drop table %s", tableName));
+    }
+
+    protected void assertValues(String tableName, int scale, List<String> expected)
+    {
+        String formattedValues = Joiner.on(", ").join(expected.stream().map(s -> format("'%s'", s)).collect(toImmutableList()));
+        assertQuery(format("select format('%%.%df', value) FROM %s", scale, tableName), "VALUES " + formattedValues);
```

Comment:
- In theory `format` may mask problems. Just `SELECT *` would be better.
- Since you have expected  values at hand, "VALUES " doesn't really help you (joiner, ,etc). you can `computeActual` and you should get BigDecimals in return